File: gpt-3.5-ft-large/template-context-full/kernels_construct_async_clause.c
#include "acc_testsuite.h"
#ifndef T1
//T1:kernels construct async clause,V:2.7-3.3
int test1(){
    int err = 0;
    srand(SEED);
    real_t a = acc_create(host_pinned, function, 1);
    real_t b = acc_create(host_pinned, function, 1);
    { 
      int _n = 1000;
        #pragma acc parallel loop copy(a[0:_n]) async(1)
        for (int i = 0; i < _n; ++i){
          a[i] = 1.0;
        }
        #pragma acc kernels loop copy(a[0:_n]) async(1)
        for (int i = 0; i < _n; ++i){
          a[i] += 1.0;
        }
      }
      while (acc_async_test(1) == acc_async_processing){
      }

      for (int x = 0; x < 1000; ++x){
          if (fabs(a[x]/b[x] - 2.0) > PRECISION){
              err = 1;
        break;
            }
      }

    return err;
}
#endif

int main(){
    int failcode = 0;
    int failed;
    int direction = 1;
#ifndef T1
    failed = 0;
    for (int x = 0; x < NUM_TEST_CALLS; ++x){
        failed = failed + test1();
    }
    if (failed != 0){
        failcode = failcode + (1 << 0);
    }
#endif
    return failcode;
}

File: gpt-3.5-ft-large/stage2/kernels_construct_async_clause.f90
#ifndef T1
!T1:kernels construct async clause,V:2.7-2.3
      LOGICAL FUNCTION test1()
        USE OPENACC
        IMPLICIT NONE
        INCLUDE "acc_testsuite.Fh"

        INTEGER :: errors = 0
        REAL(8),DIMENSION(LOOPCOUNT):: a, b, c
        INTEGER:: i, x
        LOGICAL :: AsyncsDevice1Present, AsyncsDevice2Present

        !Initilization
        SEEDDIM(1) = 1
#       ifdef SEED
        SEEDDIM(1) = SEED
#       endif
        CALL RANDOM_SEED(PUT=SEEDDIM)

        CALL RANDOM_NUMBER(a)
        CALL RANDOM_NUMBER(b)
        c = 0

        !Determine if Async is supported
        AsyncsDevice1Present = .FALSE.
        AsyncsDevice2Present = .FALSE.
        !$acc enter data copyin(c(1:LOOPCOUNT))
        !$acc data present(c(1:LOOPCOUNT)) copyout(c(1:LOOPCOUNT))
          !$acc parallel
            !$acc loop
            DO i = 1, LOOPCOUNT
              c(i) = c(i) + a(i) + b(i)
            END DO
          !$acc end parallel
        !$acc end data
        !$acc exit data delete(c(1:LOOPCOUNT))

        !$acc parallel host(AsyncsDevice1Present, AsyncsDevice2Present) 
          !$acc loop gang
          DO x = 1, 1
            !$acc loop worker
            DO i = 1, LOOPCOUNT
              c(i) = 1
            END DO
          END DO
        !$acc end parallel

        !$acc enter data copyin(a(1:LOOPCOUNT), b(1:LOOPCOUNT))
        !$acc data present(a(1:LOOPCOUNT), b(1:LOOPCOUNT)) copyout(c(1:LOOPCOUNT))
          !$acc parallel 
            !$acc loop
            DO i = 1, LOOPCOUNT
	      c(i) = c(i) + a(i) + b(i)
            END DO 
          !$acc end parallel
        !$acc end data
        a(1) = 0
        b(1) = 0

        !$acc data copyin(a(1:LOOPCOUNT), b(1:LOOPCOUNT)) copyout(c(1:LOOPCOUNT))
          !$acc parallel
            !$acc loop
            DO i = 1, LOOPCOUNT
              c(i) = c(i) - (a(i) + b(i))
            END DO
          !$acc end parallel
        !$acc end data

        CALL compare(c(1:LOOPCOUNT), a, b, c_mult, errors)

        test1 = errors
      END
#endif

      PROGRAM main
        IMPLICIT NONE
        INTEGER :: failcode, testrun
        LOGICAL :: failed
        INCLUDE "acc_testsuite.Fh"
#ifndef T1
        LOGICAL :: test1
#endif
        failed = .FALSE.
        failcode = 0
#ifndef T1
        DO testrun = 1, NUM_TEST_CALLS
          failed = failed .or. test1()
        END DO
        IF (failed) THEN
          failcode = failcode + 2 ** 0
          failed = .FALSE.
        END IF
#endif
        CALL EXIT (failcode)
      END PROGRAM